From 97b7c038f30e374116babb1531d44fa3362dc9cd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 2 Feb 2017 05:54:28 +0000
Subject: [PATCH] source/simple: Use *connect* timeout, not timeout

This ensures we only wait for the connection time, and not force the
entire build to happen within a 2 minute period. Much derp.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/builder/source/simple.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/builder/source/simple.go b/src/builder/source/simple.go
index 41d540c..27e0e01 100644
--- a/src/builder/source/simple.go
+++ b/src/builder/source/simple.go
@@ -155,7 +155,8 @@ func (s *SimpleSource) downloadCurl(destination string) error {
 	hnd.Setopt(curl.OPT_WRITEFUNCTION, writer)
 	hnd.Setopt(curl.OPT_NOPROGRESS, false)
 	hnd.Setopt(curl.OPT_PROGRESSFUNCTION, progress)
-	hnd.Setopt(curl.OPT_TIMEOUT, 120)
+	// Enforce internal 300 second connect timeout in libcurl
+	hnd.Setopt(curl.OPT_CONNECTTIMEOUT, 0)
 	hnd.Setopt(curl.OPT_USERAGENT, fmt.Sprintf("solbuild 1.3.0"))
 
 	pbar.Start()
-- 
2.11.0

